Zend/zend_language_parser.y

```diff
@@ -578,9 +578,14 @@ function_name:
 
 function_declaration_statement:
 	function returns_ref function_name backup_doc_comment '(' parameter_list ')' return_type
-	backup_fn_flags '{' inner_statement_list '}' backup_fn_flags
-		{ $$ = zend_ast_create_decl(ZEND_AST_FUNC_DECL, $2 | $13, $1, $4,
-		      zend_ast_get_str($3), $6, NULL, $11, $8, NULL); CG(extra_fn_flags) = $9; }
+	backup_fn_flags function_body backup_fn_flags
+		{ $$ = zend_ast_create_decl(ZEND_AST_FUNC_DECL, $2 | $11, $1, $4,
+		      zend_ast_get_str($3), $6, NULL, $10, $8, NULL); CG(extra_fn_flags) = $9; }
```

Comment:
This does not look correct. `function foo() = 'bar';` will result in the equivalent of `function foo() { 'bar'; return null; }`, which doesn't have the desired effect.